opendocdb-cts/internal/mongosh/fixture_test.go

```diff
@@ -162,9 +178,76 @@ func TestConvertFixtures(t *testing.T) { //nolint:revive // exceeds number of li
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
-			actual, err := ConvertFixtures(tc.fixtures)
+			t.Parallel()
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			ctx := context.Background()
+			l := slogt.New(t)
+			dbName := strings.ReplaceAll(t.Name(), "/", "-")
+
+			actualJS, err := ConvertFixtures(tc.fixtures)
+			require.NoError(t, err)
+			assert.Equal(t, unindent(tc.expected)+"\n", actualJS)
+
+			mountPath := filepath.Join(mountDir, name+".js")
+			f, err := os.Create(mountPath)
+			require.NoError(t, err)
+
+			_, err = f.WriteString(actualJS)
+			require.NoError(t, err)
+
+			require.NoError(t, f.Close())
+
+			conn, err := wireclient.Connect(ctx, "mongodb://127.0.0.1:27001/", l)
 			require.NoError(t, err)
-			assert.Equal(t, unindent(tc.expected)+"\n", actual)
+
+			t.Cleanup(func() {
+				require.NoError(t, conn.Close())
+				_ = f.Close()
+				_ = os.Remove(mountPath)
+			})
+
+			require.NoError(t, conn.Ping(ctx))
+
+			_, _, err = conn.Request(ctx, wire.MustOpMsg(
+				"dropDatabase", int32(1),
+				"$db", dbName,
+			))
+			require.NoError(t, err)
+
+			cmd := exec.Command(
+				"docker", "compose", "exec", "-T", "mongodb",
+				"mongosh", "--file", filepath.Join(containerDir, name+".js"), "mongodb://127.0.0.1:27001/"+dbName,
+			)
+			l.Debug(fmt.Sprintf("Running %s", strings.Join(cmd.Args, " ")))
+
+			err = cmd.Run()
+			require.NoError(t, err, "%s failed: %s", strings.Join(cmd.Args, " "), err)
+
+			for collName := range tc.fixtures {
+				_, body, err := conn.Request(ctx, wire.MustOpMsg(
+					"find", collName,
+					"$db", dbName,
+					// we don't expect large batch here
+					"singleBatch", true,
+				))
+				require.NoError(t, err)
+
+				doc, err := body.(*wire.OpMsg).DocumentDeep()
+				require.NoError(t, err)
+
+				//nolint:revive // don't need to check type cast
+				fixtures := doc.Get("cursor").(*wirebson.Document).Get("firstBatch").(*wirebson.Array)
+
+				var actual []*wirebson.Document
+				for v := range fixtures.Values() {
+					actual = append(actual, v.(*wirebson.Document))
+				}
+
+				assert.Equal(t, []*wirebson.Document(tc.fixtures[collName]), actual)
```

Comment:
https://github.com/FerretDB/wire/blob/e04eafb8cc3f5845485eac08d90e5a5b3d4a4c64/wiretest/wiretest.go#L70

It handles NaNs, among other things

Reply:
Just pushed v0.0.20 to make it easier to use